src/utilities/createRouterPush.ts

```diff
@@ -0,0 +1,39 @@
+import { Resolved, Route, RouterPush, RouterPushOptions, Routes } from '@/types'
+import { flattenParentMatches } from '@/utilities/flattenParentMatches'
+import { RouterNavigation } from '@/utilities/routerNavigation'
+import { component } from '@/utilities/testHelpers'
+import { assembleUrl } from '@/utilities/urlAssembly'
+
+type DummyRoutes = [{ name: string, path: '/', component: typeof component }]
```

Comment:
Curious what you think of this. I'm using this below in the push implementation on line 16 so that typescript knows the correct types for the arguments and return types. I cannot use just `RouterPush<[]>` because an empty array produces no type for the overload and then it says `string` is the only type for `urlOrRouteConfig`.

Reply:
we can't use `Route`? Is this intended to be a temporary fix?

Reply:
Cannot use Route or Routes because they both have generics that cause the infinite depth errors. But something fixed that is more narrow works just fine. 

Not intended to be temporary. Its gross. And open to other suggestions. But this is the most type safe I could make this.

Reply:
that makes sense. The things I really don't like about it is the name and the fact that it uses component from our utilities/testHelpers. 

What if we changed the type to

```ts
type AnyRoute = [{ name: string, path: string, component: RouteComponent }]
```

Reply:
Yep that totally work. Good suggestion